### PR TITLE
feat: User Management modal changes (WEB-727)

### PR DESCRIPTION
**Pull request by johnkim-det**
## Description

https://determinedai.atlassian.net/browse/WEB-727

- Split "Create User" modal into 3 separate modals (Add/View/Edit, Manage Groups, Configure Agent)
- Layout/text changes
- Minor changes to user actions dropdown

## Test Plan

with RBAC/EE:

- [ ] "Add User" modal can create a new user
- [ ] "Edit User" modal can modify user info
- [ ] "View User" action is only available when "Edit User" is not. Shows a read-only version of the fields in the "Edit User" modal
- [ ] Add/Edit/View User modals should display labels and values for each field on separate lines instead of showing label on the left and value on the right
- [ ] "Manage Groups" modal displays user's groups and allows adding to/removing from groups
- [ ] "Configure Agent" modal can modify agent info

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

[WEB-727]: https://determinedai.atlassian.net/browse/WEB-727?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/hooks/useModal/UserSettings/useModalAddToGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  return (
+    <Spinner spinning={!groups}>
+      <Form form={form} labelCol={{ span: 24 }}>
+        <Form.Item name={FIELD_NAME}>
+          <Select
+            mode="multiple"
+            optionFilterProp="children"
+            placeholder="Select Groups"
+            showSearch>
+            {groups.map((u) => (
```
Comment (gt2345) [inline webui/react/src/hooks/useModal/UserSettings/useModalAddToGroup.tsx:37]: Should filter on group, adding user to it's own group throws an error

File: webui/react/src/hooks/useModal/UserSettings/useModalCreateUser.tsx [hunk 2]
```diff
@@ -112,69 +85,33 @@ const ModalForm: React.FC<Props> = ({ form, user, groups, viewOnly, roles }) =>
...
+                  name="info"
+                  size="small"
+                  title="This user role will be applied for all workspaces and areas across the cluster."
+                />
+              </span>
+            }
+            name={ROLE_NAME}>
             <Select
               disabled={(user !== undefined && roles === null) || viewOnly}
-              mode="multiple"
```
Comment (gt2345) [inline webui/react/src/hooks/useModal/UserSettings/useModalCreateUser.tsx:171]: I wonder why are we removing multiple for global roles? Also if we change staff here, we need to modify how we call API

File: webui/react/src/hooks/useModal/UserSettings/useModalConfigureAgent.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      });
+    } else {
+      form.setFieldsValue({
+        agentGid: undefined,
+        agentGroup: undefined,
+        agentUid: undefined,
+        agentUser: undefined,
+      });
+    }
+  }, [form, user]);
```
Comment (gt2345) [inline webui/react/src/hooks/useModal/UserSettings/useModalConfigureAgent.tsx:35]: Modal value seems not updated when I modify value and reopen modal again

File: webui/react/src/hooks/useModal/UserSettings/useModalCreateUser.tsx (hunk 2 above)
Comment (johnkim-det) [reply webui/react/src/hooks/useModal/UserSettings/useModalCreateUser.tsx:171]: discussed with @alexjdetermined, removing this from the ticket.

File: webui/react/src/hooks/useModal/UserSettings/useModalAddToGroup.tsx (hunk 1 above)
Comment (johnkim-det) [reply webui/react/src/hooks/useModal/UserSettings/useModalAddToGroup.tsx]: ended up addressing this by fetching the user's groups

**Comment by johnkim-det** [comment]
recent changes in light of discussion with @alexjdetermined:
- Make "Add to Groups" display existing groups and allow removing a user from groups. Rename action and modal to "Manage Groups"
- Move “Deactivate”/“Activate” to bottom of user actions
- Rename “View Profile” to “View User” and only show when “Edit User” is not shown

**Review by gt2345 — APPROVED** [review]
LGTM, thanks for the fix

**Pull request merged**